internal/pkg/status/status.go

```diff
@@ -14,41 +14,175 @@ limitations under the License.
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cli-experimental/internal/pkg/client"
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
 	"sigs.k8s.io/cli-experimental/internal/pkg/clik8s"
 )
 
+// Condition types
+const (
+	// Level Conditions
+
+	// ConditionReady - Indicates the object is resource for use
+	ConditionReady ConditionType = "Ready"
+	// ConditionSettled - Indicates the controller is done reconciling the spec
+	// This is not implemented yet
+	ConditionSettled ConditionType = "Settled"
+
+	// Terminal condition
+
+	// ConditionFailed - The resource is in failed condition and the controller will not process it further
+	ConditionFailed ConditionType = "Failed"
+	// ConditionCompleted - The resource is done doing what it intends. Example Job, Pods can have completed state.
+	ConditionCompleted ConditionType = "Completed"
+	// Terminating - Indicates the resource is being deleted.
+	ConditionTermination ConditionType = "Terminating"
+
+	// Progress condition
+
+	// ConditionProgress - Indicates the controller is still working to satisfy the intent in the resource spec.
+	ConditionProgress ConditionType = "Progress"
+)
+
 // Status returns the status for rollouts
 type Status struct {
+	// DynamicClient is the client used to talk
+	// with the cluster
+	DynamicClient client.Client
+	// Out stores the output
+	Out io.Writer
+	// Resources is a list of resource configurations
 	Resources clik8s.ResourceConfigs
-	Out       io.Writer
-	Clientset *kubernetes.Clientset
-	Commit    *object.Commit
+	// Commit is a git commit object
+	Commit *object.Commit
+}
+
+// ConditionType - condition types
+type ConditionType string
+
+// Condition - condition object computed by status package
+type Condition struct {
+	// Type - condition type
+	Type ConditionType
+	// Status - String that describes the condition status
+	Status string // metav1.ConditionStatus
+	// Reason - Human readable string
+	Reason string
+}
+
+// ResourceStatus - resource status
+type ResourceStatus struct {
+	// Resource - unstructured object whose resource is being described
+	Resource *unstructured.Unstructured // Deletion in progress
+	// Conditions - list of extracted conditions from Resource
+	Conditions []Condition
+	// Errror - Any error encountered extracting status for this Resource
+	Error error
 }
 
 // Result contains the Status Result
 type Result struct {
-	Resources clik8s.ResourceConfigs
+	// Resources - list of resource status
+	Resources []ResourceStatus
+}
+
+// NewFalseCondition condition
+func NewFalseCondition(ctype ConditionType) Condition {
+	return Condition{Type: ctype, Status: "False", Reason: ""}
+}
+
+// NewCondition condition
+func NewCondition(ctype ConditionType, reason string) *Condition {
+	return &Condition{Type: ctype, Status: "True", Reason: reason}
+}
+
+// Get - simple wrapper to avoid *
+func (c *Condition) Get() Condition {
```

Comment:
These don't improve the readability, IMO (see my comment above).

Reply:
removed.